src/CiDetector.php

```diff
@@ -23,6 +23,16 @@ class CiDetector
     const CI_TEAMCITY = 'TeamCity';
     const CI_TRAVIS = 'Travis CI';
 
+    /**
+     * @var Env
+     */
+    private $environment;
+
+    public function __construct(Env $environment = null)
```

Comment:
I don't like to pollute the default constructor with this optional parameter - which is something you'd probably only use in tests.

I'm thinking about using named constructor for these cases, like this:

```php
    public function __construct()
    {
        $this->environment = new Env();
    }

    public function createFromEnvironment(Env $environment)
    {
        $ciDetector = new self();
        $ciDetector->environment = $environment;

        return $ciDetector;
    }
```

Reply:
Adjusted accordingly!